templateargs/argparser/arg.rl

```diff
@@ -0,0 +1,64 @@
+package argparser
+
+import "fmt"
+
+
+%%{
+    machine envvar;
+    write data;
+}%%
+
+// Var is the parsed envvar string.
+type Var struct {
+  Name  string
+  Value string
+}
+
+// Parse parses strings in the format, ${NAME} and ${NAME:VALUE}.
```

Comment:
Add a Lit struct and return LitOrVar. Lit should produce a string with `\$ => $` and `\\ => \`.